src/Fixer/ArrayNotation/AlphabeticalArrayKeySortFixer.php

```diff
@@ -0,0 +1,312 @@
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(T_ARRAY) || $tokens->isTokenKindFound(CT::T_ARRAY_SQUARE_BRACE_OPEN);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        $this->sortTokens($tokens);
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('sort_special_key_mode', 'Whether to sort the specials keys on the bottom `special_case_on_bottom` or top `special_case_on_top`.'))
+                ->setAllowedValues(['special_case_on_bottom', 'special_case_on_top'])
+                ->setDefault('special_case_on_bottom')
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * Sort the tokens in the correct order.
+     */
+    protected function sortTokens(Tokens $tokens): void
+    {
+        $lastProcessedIndex = null;
+        foreach ($tokens as $index => $token) {
+            if (null !== $lastProcessedIndex && $index < $lastProcessedIndex) {
+                continue;
+            }
+
+            if ($token->isGivenKind(T_ARRAY) || $token->isGivenKind(CT::T_ARRAY_SQUARE_BRACE_OPEN)) {
+                $clonedTokens = clone $tokens;
+
+                [$startIndex, $endIndex] = $this->getArrayIndexes($tokens, $index);
+
+                $content = [];
+                while (null !== $startIndex && $startIndex < $endIndex) {
+                    $startIndex = $tokens->getNextMeaningfulToken($startIndex);
+                    if ($tokens[$startIndex]->isGivenKind(T_ARRAY) || $tokens[$startIndex]->isGivenKind(CT::T_ARRAY_SQUARE_BRACE_OPEN)) {
+                        $lastProcessedIndex = $startIndex = $this->sortNestedTokens($clonedTokens, $startIndex);
+                    }
+
+                    if ($tokens[$startIndex]->isGivenKind(T_DOUBLE_ARROW)) {
+                        [$key, $keyTokenIndex] = $this->getKeyAndEndPosition($clonedTokens, $startIndex);
+
+                        $valueTokenIndex = $tokens->getNextMeaningfulToken($startIndex);
+
+                        if ($tokens[$valueTokenIndex]->isGivenKind(T_ARRAY) || $tokens[$valueTokenIndex]->isGivenKind(CT::T_ARRAY_SQUARE_BRACE_OPEN)) {
+                            $startIndex = $this->sortNestedTokens($clonedTokens, $valueTokenIndex);
+                        }
+
+                        $valueRange = $startIndex;
+                        while ($valueRange = $tokens->getNextTokenOfKind($valueRange, [',', '('])) {
+                            if ($tokens[$valueRange]->equals('(')) {
+                                $valueRange = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $valueRange);
+                            } else {
+                                break;
+                            }
+                        }
+
+                        if (!$valueRange || $valueRange > $endIndex) {
```

Comment:
```suggestion
                        if (0 !== $valueRange || $valueRange > $endIndex) {
```
No boolish magic please. Is the part `!$valueRange` even needed?